pkg/v0/cmd.go

```diff
@@ -160,9 +169,94 @@ func Run(ctx context.Context, logger *logrus.Logger, opts Options) error {
 	return nil
 }
 
+func getTagOrCommit(ctx context.Context, repo string, opts Options, logger *logrus.Entry) (string, error) {
+	module := fmt.Sprintf("github.com/%s", repo)
+	rawInfo, err := internal.RunCommand(logger, internal.WithDir(exec.CommandContext(ctx,
+		"go", "list", "-json", "-m", module), "staging/operator-lifecycle-manager"))
```

Comment:
No very familiar with this repo so I'm trying to make sure I understand the flow that is taken here. 

With this using the `staging/operator-lifecycle-manager` directory assumes that the upstream OLM `master` change has already been downstreamed or it would result in the dependency repos being behind by one sync.

The way I'm currently understanding this is that the upstream OLM `master` branch will be synced followed by the api + o-reg branches/versions on the next sync - is this accurate?

Reply:
You actually make a good point, and we haven't been able to test this scenario. The operator-lifecycle-manager repo is fetched, and then the other repos are fetched - based on the current state of this directory. Nothing has been merged/synced. So out-of-date data _could_ be used. Then the OLM changes are synced without the corresponding API or operator-registry changes, and things blow up.

So, I _think_ what needs to happen is:
1. Fetch the OLM repo.
2. Look for new commits on the OLM repo.
3. Merge the next commit from the OLM repo.
4. fetch the API and operator-registry repos based on `staging/operator-lifecycle-manager`
5. Merge those API and operator-registry commits.
6. Run `go mod` commands
7. Run `make manifest`
8. Goto step 3.

Because, now that I think about it, the updates from API and operator-registry are ignored because `go list` is used at the wrong time.